controllers/cloud.redhat.com/providers/kafka/strimzi.go

```diff
@@ -297,18 +328,88 @@ func (s *strimziProvider) Provide(app *crd.ClowdApp, c *config.AppConfig) error
 		}
 	}
 
+	ku := &strimzi.KafkaUser{}
+	nn := types.NamespacedName{
+		Name:      getKafkaUsername(s.Env, app),
+		Namespace: getKafkaNamespace(s.Env),
+	}
+
+	s.Cache.Create(KafkaUser, nn, ku)
+	labeler := utils.GetCustomLabeler(
+		map[string]string{"strimzi.io/cluster": s.Env.Spec.Providers.Kafka.Cluster.Name}, nn, s.Env,
+	)
+
+	labeler(ku)
+
+	ku.Spec = &strimzi.KafkaUserSpec{
+		Authentication: &strimzi.KafkaUserSpecAuthentication{
+			Type: strimzi.KafkaUserSpecAuthenticationTypeScramSha512,
+		},
+		Authorization: &strimzi.KafkaUserSpecAuthorization{
+			Acls: []strimzi.KafkaUserSpecAuthorizationAclsElem{},
+			Type: strimzi.KafkaUserSpecAuthorizationTypeSimple,
+		},
+	}
+
 	// update s.Config.Topics
-	if err := s.processTopics(app); err != nil {
+	if err := s.processTopics(app, ku); err != nil {
 		return err
 	}
 
+	s.Cache.Update(KafkaUser, ku)
+
+	for _, topic := range s.Config.Brokers {
```

Comment:
I'm sorry but I cannot understand how iterating over `Brokers` gives you a set of `topic` objects.  This makes it impossible to understand what this block of code is doing.

Reply:
You are absolutely right - I can't understand how I did that...